src/ddsrt/src/library/posix/library.c

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright(c) 2006 to 2018 ADLINK Technology Limited and others
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Eclipse Distribution License
+ * v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+#include <stdio.h>
+#include <dlfcn.h>
+#include <assert.h>
+#include <string.h>
+#include "dds/ddsrt/heap.h"
+#include "dds/ddsrt/library.h"
+
+
+dds_retcode_t
+ddsrt_dlopen(
+    const char *name,
+    bool translate,
+    ddsrt_lib *handle)
+{
+  dds_retcode_t retcode = DDS_RETCODE_OK;
+
+
+  if ( name == NULL || handle == NULL ){
+      retcode = DDS_RETCODE_BAD_PARAMETER;
+
+  } else {
+      *handle = NULL;
+      dlerror();    /* Clear any existing error */
+      if ( (translate) && (strrchr(name, '/') == NULL)) {
+        /* Add lib and suffix to the name and try to open. */
+#if __APPLE__
+        static const char suffix[] = ".dylib";
+#else
+        static const char suffix[] = ".so";
+#endif
+        char* libName = ddsrt_malloc(3 + strlen(name) + sizeof(suffix));
+        sprintf(libName, "lib%s%s", name, suffix);
```

Comment:
Better not use `sprintf` here, but `snprintf`. If only to avoid compiler warnings. `ddsrt_asprintf` would be an option to, that way you can also ditch the `ddsrt_malloc`.

Reply:
As this would be the _only_ use of ``sprintf`` ... yes, please replace it.